docs/troubleshooting/unable-to-install-workload.md

```diff
@@ -0,0 +1,24 @@
+---
+title: Troubleshoot installing the .NET Aspire workload
+description: Explore strategies for troubleshooting issues when installing the .NET Aspire workload.
+ms.date: 05/15/2024
+---
+
+# Troubleshoot installing the .NET Aspire workload
+
+This article provides guidance on how to troubleshoot issues that you might encounter when installing the .NET Aspire workload.
+
+## Symptoms
+
+When you install the .NET Aspire workload, you might encounter an installation error. The error message might indicate that the installation failed, or that the workload couldn't be installed. The error message might also indicate that a package source is unavailable, or that a package source isn't found.
+
+## Possible solution
+
+Ensure that any recursive _Nuget.config_ files are configured to specify the correct package sources and NuGet feeds. For example, if you have a _Nuget.config_ file in your user profile directory, ensure that it doesn't specify a package source that is no longer available.
+
+<!-- TODO: Ask Jose to help fill in more details, and perhaps even more symptoms. -->
```

Comment:
First, to help set some context, workload manifests are essentially some pre-knowledge that the SDK has of a workload whether or not you have it installed. So, for instance, even if you don't have Maui workload installed in your machine, your SDK still has a workload manifest to help it be aware that it exists.

The bulk of the issues people hit when installing workloads is due to the same root cause: Somehow your SDK is aware of some workload manifest or workload pack versions that are not present in any of the feeds configured when you are trying to run the `dotnet workload` commands. This can happen for different reasons:

- [Most common] The SDK will, once a day, try to do a search to see if there are any updates on any of the workload manifests. The way it happens, is that when you run your first `dotnet` command that day, for instance running `dotnet build` in a repo, it will use this dotnet invocation to spin off a background process to check if there are any updates to the workload manifests. The main issue, is that it checks using the feeds that this particular dotnet invocation had, so if for example, you had the dotnet8 feed as one of the feeds used in that repo you were building, well it now will try to look in dotnet8 feed to see if it finds updates for any of your manifests, and if found, it will just save some state in the SDK to know that this version exists. The problem then, is that the next time you run `dotnet workload update` or `dotnet workload install blah`, it now knows that there is a new version of the manifest for one of the workloads, but it is possible that these workload commands are using a different set of feeds (meaning a different NuGet.config) and now it can't find that version it found previously, so it errors out. When you are in this boat, I believe that there isn't an easy way to "forget" this state of the new version, and so you basically need to find the feed where that version of the manifest is located. This is why sometimes even when running a command like `dotnet workload install aspire` fails saying that it cannot find some xamarin android package, even when all you care about is aspire and you may not even have the xamarin workload installed. Customers not working with our feeds where we push these manifests to (like the dotnet8 feed) are of course less prone to hit these errors.

- [Not so common, but still happens] Even if you are using the right feeds (meaning the same feeds used when that daily check found a newer version), there are cases where one workload manifest will have a dependency to a workload pack that is not published on the same feed, so you can hit errors where now you need to find the feed where that pack is located.